src/Plugins/DBFTPlugin/Consensus/ConsensusService.cs

```diff
@@ -320,7 +320,7 @@ private void ChangeTimer(TimeSpan delay)
         private void ExtendTimerByFactor(int maxDelayInBlockTimes)
         {
             TimeSpan nextDelay = expected_delay - (TimeProvider.Current.UtcNow - clock_started)
-                + TimeSpan.FromMilliseconds(maxDelayInBlockTimes * neoSystem.Settings.MillisecondsPerBlock / (double)context.M);
+                + TimeSpan.FromMilliseconds(maxDelayInBlockTimes * context.TimePerBlock.TotalMilliseconds / (double)context.M);
```

Comment:
Later we need to improve this formula as @superboyiii said.

This requires larger extend by factor when block times are slower because the NETWORK P2P delay did not change (it is the same for both cases).
Here we have a parameter that is based on the network delay basically but that was calibrated for 15s block.

Let's keep this in mind @shargon @AnnaShaleva @Jim8y @roman-khimov

Reply:
It's a separate issue that needs careful evaluation, so it won't be done in the current PR.